File: Java/Problem_073.java
public class Euler73 {
    static int gcd(int a, int b) {
        while (b != 0) {
            int t = a % b;
            a = b;
            b = t;
        }
        return a;
    }

    public static void main(String[] args) {
        final int LIMIT = 12000;
        long count = 0;

        for (int d = 2; d <= LIMIT; d++) {
            int nStart = d / 3 + 1;
            int nEnd = (d - 1) / 2;
            for (int n = nStart; n <= nEnd; n++) {
                if (gcd(n, d) == 1) count++;
            }
        }

        System.out.println(count); // 7295372
    }
}

File: Python/Problem_073.py
import math

def euler_73(limit: int = 12000) -> int:
    count = 0
    for d in range(2, limit + 1):
        n_start = d // 3 + 1
        n_end = (d - 1) // 2
        for n in range(n_start, n_end + 1):
            if math.gcd(n, d) == 1:
                count += 1
    return count

if __name__ == "__main__":
    print(euler_73())  # 7295372
